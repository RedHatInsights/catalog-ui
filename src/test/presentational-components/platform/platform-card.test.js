import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import PlatformCard from '../../../presentational-components/platform/platform-card';

describe('<PlatformCard />', () => {
  let initialProps;
  beforeEach(() => {
    initialProps = {
      description: 'desc',
      modified: 'Foo',
      id: '123'
    };
  });

  it('should render correctly', () => {
    const wrapper = mount(
      <MemoryRouter>
        <PlatformCard { ...initialProps } />
      </MemoryRouter>
    );
    expect(toJson(wrapper.find(PlatformCard))).toMatchSnapshot();
  });

  it('should choose card image', () => {
    let wrapper = mount(
      <MemoryRouter>
        <PlatformCard { ...initialProps } />
      </MemoryRouter>
    );
    expect(toJson(wrapper.find(PlatformCard))).toMatchSnapshot();
  });
});
