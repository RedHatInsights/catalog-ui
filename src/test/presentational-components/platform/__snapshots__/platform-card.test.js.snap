// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlatformCard /> should choose card image 1`] = `
<PlatformCard
  description="desc"
  modified="Foo"
>
  <GalleryItem>
    <div>
      <Link
        className="card-link"
        to="/platforms/detail/undefined"
      >
        <LinkAnchor
          className="card-link"
          href="/platforms/detail/undefined"
          navigate={[Function]}
        >
          <a
            className="card-link"
            href="/platforms/detail/undefined"
            onClick={[Function]}
          >
            <Card
              className="content-gallery-card"
            >
              <article
                className="pf-c-card content-gallery-card"
              >
                <CardHeader>
                  <div
                    className="pf-c-card__header pf-c-title pf-m-md"
                  >
                    <PlatformCardHeader
                      headerActions={
                        Array [
                          <Dropdown
                            dropdownItems={
                              Array [
                                <DropdownItem
                                  component={
                                    <ForwardRef
                                      to="/platforms/edit-workflow/undefined"
                                    >
                                      Edit approval
                                    </ForwardRef>
                                  }
                                />,
                              ]
                            }
                            isOpen={false}
                            isPlain={true}
                            onSelect={[Function]}
                            position="right"
                            toggle={
                              <KebabToggle
                                onToggle={[Function]}
                              />
                            }
                          />,
                        ]
                      }
                      image="test-file-stub"
                    >
                      <Level>
                        <div
                          className="pf-l-level"
                        >
                          <ImageWithDefault
                            height="40"
                            src="test-file-stub"
                            width="80"
                          >
                            <img
                              height="40"
                              src="test-file-stub"
                              width="80"
                            />
                          </ImageWithDefault>
                          <LevelItem
                            onClick={[Function]}
                          >
                            <div
                              onClick={[Function]}
                            >
                              <Dropdown
                                dropdownItems={
                                  Array [
                                    <DropdownItem
                                      component={
                                        <ForwardRef
                                          to="/platforms/edit-workflow/undefined"
                                        >
                                          Edit approval
                                        </ForwardRef>
                                      }
                                    />,
                                  ]
                                }
                                isOpen={false}
                                isPlain={true}
                                key="portfolio-dropdown"
                                onSelect={[Function]}
                                position="right"
                                toggle={
                                  <KebabToggle
                                    onToggle={[Function]}
                                  />
                                }
                              >
                                <Component
                                  dropdownItems={
                                    Array [
                                      <DropdownItem
                                        component={
                                          <ForwardRef
                                            to="/platforms/edit-workflow/undefined"
                                          >
                                            Edit approval
                                          </ForwardRef>
                                        }
                                      />,
                                    ]
                                  }
                                  isOpen={false}
                                  isPlain={true}
                                  position="right"
                                  toggle={
                                    <KebabToggle
                                      onToggle={[Function]}
                                    />
                                  }
                                >
                                  <ComponentWithOuia
                                    component={[Function]}
                                    componentProps={
                                      Object {
                                        "dropdownItems": Array [
                                          <DropdownItem
                                            component={
                                              <ForwardRef
                                                to="/platforms/edit-workflow/undefined"
                                              >
                                                Edit approval
                                              </ForwardRef>
                                            }
                                          />,
                                        ],
                                        "isOpen": false,
                                        "isPlain": true,
                                        "position": "right",
                                        "toggle": <KebabToggle
                                          onToggle={[Function]}
                                        />,
                                      }
                                    }
                                    consumerContext={null}
                                  >
                                    <DropdownWithContext
                                      autoFocus={true}
                                      className=""
                                      direction="down"
                                      dropdownItems={
                                        Array [
                                          <DropdownItem
                                            component={
                                              <ForwardRef
                                                to="/platforms/edit-workflow/undefined"
                                              >
                                                Edit approval
                                              </ForwardRef>
                                            }
                                          />,
                                        ]
                                      }
                                      isGrouped={false}
                                      isOpen={false}
                                      isPlain={true}
                                      onSelect={[Function]}
                                      ouiaComponentType="Dropdown"
                                      ouiaContext={
                                        Object {
                                          "isOuia": false,
                                          "ouiaId": null,
                                        }
                                      }
                                      position="right"
                                      toggle={
                                        <KebabToggle
                                          onToggle={[Function]}
                                        />
                                      }
                                    >
                                      <div
                                        className="pf-c-dropdown"
                                      >
                                        <KebabToggle
                                          ariaHasPopup={true}
                                          id="pf-toggle-id-1"
                                          isOpen={false}
                                          isPlain={true}
                                          key=".0"
                                          onEnter={[Function]}
                                          onToggle={[Function]}
                                          parentRef={
                                            Object {
                                              "current": <div
                                                class="pf-c-dropdown"
                                              >
                                                <button
                                                  aria-expanded="false"
                                                  aria-haspopup="true"
                                                  aria-label="Actions"
                                                  class="pf-c-dropdown__toggle pf-m-plain"
                                                  id="pf-toggle-id-1"
                                                  type="button"
                                                >
                                                  <svg
                                                    aria-hidden="true"
                                                    fill="currentColor"
                                                    height="1em"
                                                    role="img"
                                                    style="vertical-align: -0.125em;"
                                                    viewBox="0 0 192 512"
                                                    width="1em"
                                                  >
                                                    <path
                                                      d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                                                      transform=""
                                                    />
                                                  </svg>
                                                </button>
                                              </div>,
                                            }
                                          }
                                        >
                                          <Toggle
                                            aria-label="Actions"
                                            ariaHasPopup={true}
                                            className=""
                                            id="pf-toggle-id-1"
                                            isActive={false}
                                            isDisabled={false}
                                            isFocused={false}
                                            isHovered={false}
                                            isOpen={false}
                                            isPlain={true}
                                            isPrimary={false}
                                            isSplitButton={false}
                                            onEnter={[Function]}
                                            onToggle={[Function]}
                                            parentRef={
                                              Object {
                                                "current": <div
                                                  class="pf-c-dropdown"
                                                >
                                                  <button
                                                    aria-expanded="false"
                                                    aria-haspopup="true"
                                                    aria-label="Actions"
                                                    class="pf-c-dropdown__toggle pf-m-plain"
                                                    id="pf-toggle-id-1"
                                                    type="button"
                                                  >
                                                    <svg
                                                      aria-hidden="true"
                                                      fill="currentColor"
                                                      height="1em"
                                                      role="img"
                                                      style="vertical-align: -0.125em;"
                                                      viewBox="0 0 192 512"
                                                      width="1em"
                                                    >
                                                      <path
                                                        d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                                                        transform=""
                                                      />
                                                    </svg>
                                                  </button>
                                                </div>,
                                              }
                                            }
                                          >
                                            <button
                                              aria-expanded={false}
                                              aria-haspopup={true}
                                              aria-label="Actions"
                                              className="pf-c-dropdown__toggle pf-m-plain"
                                              disabled={false}
                                              id="pf-toggle-id-1"
                                              onClick={[Function]}
                                              onKeyDown={[Function]}
                                              type="button"
                                            >
                                              <EllipsisVIcon
                                                color="currentColor"
                                                noVerticalAlign={false}
                                                size="sm"
                                                title={null}
                                              >
                                                <svg
                                                  aria-hidden={true}
                                                  aria-labelledby={null}
                                                  fill="currentColor"
                                                  height="1em"
                                                  role="img"
                                                  style={
                                                    Object {
                                                      "verticalAlign": "-0.125em",
                                                    }
                                                  }
                                                  viewBox="0 0 192 512"
                                                  width="1em"
                                                >
                                                  <path
                                                    d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                                                    transform=""
                                                  />
                                                </svg>
                                              </EllipsisVIcon>
                                            </button>
                                          </Toggle>
                                        </KebabToggle>
                                      </div>
                                    </DropdownWithContext>
                                  </ComponentWithOuia>
                                </Component>
                              </Dropdown>
                            </div>
                          </LevelItem>
                        </div>
                      </Level>
                    </PlatformCardHeader>
                  </div>
                </CardHeader>
                <CardBody>
                  <div
                    className="pf-c-card__body"
                  >
                    <h4 />
                    <ItemDetails
                      description="desc"
                      modified="Foo"
                      toDisplay={
                        Array [
                          "description",
                          "modified",
                        ]
                      }
                    >
                      <Dotdotdot
                        clamp={6}
                        tagName="div"
                        truncationChar="…"
                        useNativeClamp={false}
                      >
                        <div>
                          <PropLine
                            key="card-prop-description"
                            value="desc"
                          >
                            <div>
                              desc
                            </div>
                          </PropLine>
                          <PropLine
                            key="card-prop-modified"
                            value="Foo"
                          >
                            <div>
                              Foo
                            </div>
                          </PropLine>
                        </div>
                      </Dotdotdot>
                    </ItemDetails>
                  </div>
                </CardBody>
                <CardFooter>
                  <div
                    className="pf-c-card__footer"
                  />
                </CardFooter>
              </article>
            </Card>
          </a>
        </LinkAnchor>
      </Link>
    </div>
  </GalleryItem>
</PlatformCard>
`;

exports[`<PlatformCard /> should render correctly 1`] = `
<PlatformCard
  description="desc"
  modified="Foo"
>
  <GalleryItem>
    <div>
      <Link
        className="card-link"
        to="/platforms/detail/undefined"
      >
        <LinkAnchor
          className="card-link"
          href="/platforms/detail/undefined"
          navigate={[Function]}
        >
          <a
            className="card-link"
            href="/platforms/detail/undefined"
            onClick={[Function]}
          >
            <Card
              className="content-gallery-card"
            >
              <article
                className="pf-c-card content-gallery-card"
              >
                <CardHeader>
                  <div
                    className="pf-c-card__header pf-c-title pf-m-md"
                  >
                    <PlatformCardHeader
                      headerActions={
                        Array [
                          <Dropdown
                            dropdownItems={
                              Array [
                                <DropdownItem
                                  component={
                                    <ForwardRef
                                      to="/platforms/edit-workflow/undefined"
                                    >
                                      Edit approval
                                    </ForwardRef>
                                  }
                                />,
                              ]
                            }
                            isOpen={false}
                            isPlain={true}
                            onSelect={[Function]}
                            position="right"
                            toggle={
                              <KebabToggle
                                onToggle={[Function]}
                              />
                            }
                          />,
                        ]
                      }
                      image="test-file-stub"
                    >
                      <Level>
                        <div
                          className="pf-l-level"
                        >
                          <ImageWithDefault
                            height="40"
                            src="test-file-stub"
                            width="80"
                          >
                            <img
                              height="40"
                              src="test-file-stub"
                              width="80"
                            />
                          </ImageWithDefault>
                          <LevelItem
                            onClick={[Function]}
                          >
                            <div
                              onClick={[Function]}
                            >
                              <Dropdown
                                dropdownItems={
                                  Array [
                                    <DropdownItem
                                      component={
                                        <ForwardRef
                                          to="/platforms/edit-workflow/undefined"
                                        >
                                          Edit approval
                                        </ForwardRef>
                                      }
                                    />,
                                  ]
                                }
                                isOpen={false}
                                isPlain={true}
                                key="portfolio-dropdown"
                                onSelect={[Function]}
                                position="right"
                                toggle={
                                  <KebabToggle
                                    onToggle={[Function]}
                                  />
                                }
                              >
                                <Component
                                  dropdownItems={
                                    Array [
                                      <DropdownItem
                                        component={
                                          <ForwardRef
                                            to="/platforms/edit-workflow/undefined"
                                          >
                                            Edit approval
                                          </ForwardRef>
                                        }
                                      />,
                                    ]
                                  }
                                  isOpen={false}
                                  isPlain={true}
                                  position="right"
                                  toggle={
                                    <KebabToggle
                                      onToggle={[Function]}
                                    />
                                  }
                                >
                                  <ComponentWithOuia
                                    component={[Function]}
                                    componentProps={
                                      Object {
                                        "dropdownItems": Array [
                                          <DropdownItem
                                            component={
                                              <ForwardRef
                                                to="/platforms/edit-workflow/undefined"
                                              >
                                                Edit approval
                                              </ForwardRef>
                                            }
                                          />,
                                        ],
                                        "isOpen": false,
                                        "isPlain": true,
                                        "position": "right",
                                        "toggle": <KebabToggle
                                          onToggle={[Function]}
                                        />,
                                      }
                                    }
                                    consumerContext={null}
                                  >
                                    <DropdownWithContext
                                      autoFocus={true}
                                      className=""
                                      direction="down"
                                      dropdownItems={
                                        Array [
                                          <DropdownItem
                                            component={
                                              <ForwardRef
                                                to="/platforms/edit-workflow/undefined"
                                              >
                                                Edit approval
                                              </ForwardRef>
                                            }
                                          />,
                                        ]
                                      }
                                      isGrouped={false}
                                      isOpen={false}
                                      isPlain={true}
                                      onSelect={[Function]}
                                      ouiaComponentType="Dropdown"
                                      ouiaContext={
                                        Object {
                                          "isOuia": false,
                                          "ouiaId": null,
                                        }
                                      }
                                      position="right"
                                      toggle={
                                        <KebabToggle
                                          onToggle={[Function]}
                                        />
                                      }
                                    >
                                      <div
                                        className="pf-c-dropdown"
                                      >
                                        <KebabToggle
                                          ariaHasPopup={true}
                                          id="pf-toggle-id-0"
                                          isOpen={false}
                                          isPlain={true}
                                          key=".0"
                                          onEnter={[Function]}
                                          onToggle={[Function]}
                                          parentRef={
                                            Object {
                                              "current": <div
                                                class="pf-c-dropdown"
                                              >
                                                <button
                                                  aria-expanded="false"
                                                  aria-haspopup="true"
                                                  aria-label="Actions"
                                                  class="pf-c-dropdown__toggle pf-m-plain"
                                                  id="pf-toggle-id-0"
                                                  type="button"
                                                >
                                                  <svg
                                                    aria-hidden="true"
                                                    fill="currentColor"
                                                    height="1em"
                                                    role="img"
                                                    style="vertical-align: -0.125em;"
                                                    viewBox="0 0 192 512"
                                                    width="1em"
                                                  >
                                                    <path
                                                      d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                                                      transform=""
                                                    />
                                                  </svg>
                                                </button>
                                              </div>,
                                            }
                                          }
                                        >
                                          <Toggle
                                            aria-label="Actions"
                                            ariaHasPopup={true}
                                            className=""
                                            id="pf-toggle-id-0"
                                            isActive={false}
                                            isDisabled={false}
                                            isFocused={false}
                                            isHovered={false}
                                            isOpen={false}
                                            isPlain={true}
                                            isPrimary={false}
                                            isSplitButton={false}
                                            onEnter={[Function]}
                                            onToggle={[Function]}
                                            parentRef={
                                              Object {
                                                "current": <div
                                                  class="pf-c-dropdown"
                                                >
                                                  <button
                                                    aria-expanded="false"
                                                    aria-haspopup="true"
                                                    aria-label="Actions"
                                                    class="pf-c-dropdown__toggle pf-m-plain"
                                                    id="pf-toggle-id-0"
                                                    type="button"
                                                  >
                                                    <svg
                                                      aria-hidden="true"
                                                      fill="currentColor"
                                                      height="1em"
                                                      role="img"
                                                      style="vertical-align: -0.125em;"
                                                      viewBox="0 0 192 512"
                                                      width="1em"
                                                    >
                                                      <path
                                                        d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                                                        transform=""
                                                      />
                                                    </svg>
                                                  </button>
                                                </div>,
                                              }
                                            }
                                          >
                                            <button
                                              aria-expanded={false}
                                              aria-haspopup={true}
                                              aria-label="Actions"
                                              className="pf-c-dropdown__toggle pf-m-plain"
                                              disabled={false}
                                              id="pf-toggle-id-0"
                                              onClick={[Function]}
                                              onKeyDown={[Function]}
                                              type="button"
                                            >
                                              <EllipsisVIcon
                                                color="currentColor"
                                                noVerticalAlign={false}
                                                size="sm"
                                                title={null}
                                              >
                                                <svg
                                                  aria-hidden={true}
                                                  aria-labelledby={null}
                                                  fill="currentColor"
                                                  height="1em"
                                                  role="img"
                                                  style={
                                                    Object {
                                                      "verticalAlign": "-0.125em",
                                                    }
                                                  }
                                                  viewBox="0 0 192 512"
                                                  width="1em"
                                                >
                                                  <path
                                                    d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                                                    transform=""
                                                  />
                                                </svg>
                                              </EllipsisVIcon>
                                            </button>
                                          </Toggle>
                                        </KebabToggle>
                                      </div>
                                    </DropdownWithContext>
                                  </ComponentWithOuia>
                                </Component>
                              </Dropdown>
                            </div>
                          </LevelItem>
                        </div>
                      </Level>
                    </PlatformCardHeader>
                  </div>
                </CardHeader>
                <CardBody>
                  <div
                    className="pf-c-card__body"
                  >
                    <h4 />
                    <ItemDetails
                      description="desc"
                      modified="Foo"
                      toDisplay={
                        Array [
                          "description",
                          "modified",
                        ]
                      }
                    >
                      <Dotdotdot
                        clamp={6}
                        tagName="div"
                        truncationChar="…"
                        useNativeClamp={false}
                      >
                        <div>
                          <PropLine
                            key="card-prop-description"
                            value="desc"
                          >
                            <div>
                              desc
                            </div>
                          </PropLine>
                          <PropLine
                            key="card-prop-modified"
                            value="Foo"
                          >
                            <div>
                              Foo
                            </div>
                          </PropLine>
                        </div>
                      </Dotdotdot>
                    </ItemDetails>
                  </div>
                </CardBody>
                <CardFooter>
                  <div
                    className="pf-c-card__footer"
                  />
                </CardFooter>
              </article>
            </Card>
          </a>
        </LinkAnchor>
      </Link>
    </div>
  </GalleryItem>
</PlatformCard>
`;
